src/main/java/org/rumbledb/runtime/operational/CastableIterator.java

```diff
@@ -21,12 +22,12 @@
 
 public class CastableIterator extends LocalRuntimeIterator {
     private static final long serialVersionUID = 1L;
-    private final SingleType singleType;
+    private final SequenceType singleType;
```

Comment:
For consistency with CastIterator, I would recommend renaming this to sequenceType as well. Alternatively, CastIterator can keep the name singleType if it is meaningful.